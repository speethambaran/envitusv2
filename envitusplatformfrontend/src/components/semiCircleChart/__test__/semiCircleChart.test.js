import React from 'react';
import { shallow } from 'enzyme';
import SemiCircleChart from './../semiCircleChart';
import { findByDataTest } from './../../../Utils';

const preConfig = (props={}) => {
    const component = shallow(<SemiCircleChart {...props} />);
    return component;
};

describe('SemiCircleChart Test', () => {
    it('snapshot', () => {
        const wrapper = shallow(<SemiCircleChart/>);   
        expect(wrapper).toMatchSnapshot();
    });

    it('Redering Properly with prop values', () => {
        let component;
        component = preConfig();
        const wrapper = findByDataTest(component, 'semiCircleComponent');
        expect(wrapper.length).toBe(1);
    });

    it('draws Chart', () => {
        let component;
        const props = {
            value: 20,
            max: 50,
            min: 2,
            unit: "mg",
            color: "F5F5F5"
        };
        component = preConfig(props);
        component.setState({
            node: React.createRef()
        });
        expect(component.instance().render().ref.current).toBe(null);  
    })
});