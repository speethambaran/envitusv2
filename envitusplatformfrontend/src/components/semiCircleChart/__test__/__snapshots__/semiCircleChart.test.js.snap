// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SemiCircleChart Test snapshot 1`] = `
<svg
  data-test="semiCircleComponent"
/>
`;
